comp/languagedetection/client/client.go

```diff
@@ -341,6 +340,7 @@ func (c *client) handleProcessEvent(processEvent workloadmeta.Event, isRetry boo
 	added := containerInfo.Add(string(process.Language.Name))
 	if added {
 		c.freshlyUpdatedPods[pod.Name] = struct{}{}
+		delete(c.processesWithoutPod, process.ContainerID)
```

Comment:
We remove the process event from `processesWithoutPod` when the pod is added to `freshlyUpdatedPods` 

If the handling fails, the cleanup of `processesWithoutPods` (which runs [here](https://github.com/DataDog/datadog-agent/blob/a4dda346c4dcf0e901c348c3103f060fcdd66bca/comp/languagedetection/client/client.go#L180-L188)) will delete the process event eventually.